Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/GazeContingent/FixWindowBufferedSamples/EyeLink_FixWindowBufferedSamples.m

```diff
@@ -0,0 +1,538 @@
+    end
+    
+    % Get EyeLink tracker and software version
+    % <ver> returns 0 if not connected
+    % <versionstring> returns 'EYELINK I', 'EYELINK II x.xx', 'EYELINK CL x.xx' where 'x.xx' is the software version
+    ELsoftwareVersion = 0; % Default EyeLink version in dummy mode
+    [ver, versionstring] = Eyelink('GetTrackerVersion');
+    if dummymode == 0 % If connected to EyeLink
+        % Extract software version number. 
+        [~, vnumcell] = regexp(versionstring,'.*?(\d)\.\d*?','Match','Tokens'); % Extract EL version before decimal point
+        ELsoftwareVersion = str2double(vnumcell{1}{1}); % Returns 1 for EyeLink I, 2 for EyeLink II, 3/4 for EyeLink 1K, 5 for EyeLink 1KPlus, 6 for Portable Duo
+        % Print some text in Matlab's Command Window
+        fprintf('Running experiment on %s version %d\n', versionstring, ver );
+    end
+    % Add a line of text in the EDF file to identify the current experimemt name and session. This is optional.
+    % If your text starts with "RECORDED BY " it will be available in DataViewer's Inspector window by clicking
+    % the EDF session node in the top panel and looking for the "Recorded By:" field in the bottom panel of the Inspector.
+    preambleText = sprintf('RECORDED BY Psychtoolbox demo %s session name: %s', mfilename, edfFile);
+    Eyelink('Command', 'add_file_preamble_text "%s"', preambleText);
+    
+    
+    %% STEP 2: SELECT AVAILABLE SAMPLE/EVENT DATA
+    % See EyeLinkProgrammers Guide manual > Useful EyeLink Commands > File Data Control & Link Data Control
+    
+    % Select which events are saved in the EDF file. Include everything just in case
+    Eyelink('Command', 'file_event_filter = LEFT,RIGHT,FIXATION,SACCADE,BLINK,MESSAGE,BUTTON,INPUT');
+    % Select which events are available online for gaze-contingent experiments. Include everything just in case
+    Eyelink('Command', 'link_event_filter = LEFT,RIGHT,FIXATION,SACCADE,BLINK,BUTTON,FIXUPDATE,INPUT');
+    % Select which sample data is saved in EDF file or available online. Include everything just in case
+    if ELsoftwareVersion > 3  % Check tracker version and include 'HTARGET' to save head target sticker data for supported eye trackers
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,HTARGET,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,HTARGET,STATUS,INPUT');
+    else
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,STATUS,INPUT');
+    end
+    
+    
+    %% STEP 3: OPEN GRAPHICS WINDOW
+    
+    % Open experiment graphics on the specified screen
+    if isempty(screenNumber)
+        screenNumber = max(Screen('Screens')); % Use default screen if none specified
+    end
+    [window, wRect] = Screen('OpenWindow', screenNumber, [128 128 128]); % Open graphics window

```

Comment:
The recommendation nowadays is to generally use ``PsychImaging('OpenWindow', ...)`` instead of ``Screen('OpenWindow', ...)``, as it allows PTB to do some clever setup. E.g., if you add
``PsychDefaultSetup(n);`` as required, with a n of 2 or higher, as recommended for new code, then this will switch the color system from the old legacy 0 - 255 color range to a normalized 0.0 - 1.0 color range, where 0 is black, 1 is white, 0.5 is a 50% intensity etc. The same applies to all textures created via ``Screen('MakeTexture', win, img);`` if the img is of data type double(). This has the advantage that color specifications work regardless if you display to a standard 8 bit monitor, or to a high precision deep color 10 or 12 bit monitor, or to some widely used visual stimulator like Cambridge Research, VPixx, VideoSwitcher and others. See `help PsychDefaultSetup` for what it does for different feature levels.

Same applies to the other demos - it is good practice to show now.

Reply:
resolved in cf762efe7df1a9e7d29650dfcb83c66c0b571bd7